### PR TITLE
Fix mirroring from waypoints

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/istio/issues/52713

### CODE REVIEW

**Comment by keithmattix** [comment]
Can you explain the bug more in the description?

File: pilot/pkg/networking/core/route/route.go [hunk 1]
```diff
@@ -1387,9 +1387,14 @@ func TranslateFault(in *networking.HTTPFaultInjection) *xdshttpfault.HTTPFault {
 
 func TranslateRequestMirrorPolicy(dst *networking.Destination, service *model.Service,
 	listenerPort int, mp *core.RuntimeFractionalPercent,
+) *route.RouteAction_RequestMirrorPolicy {
+	return TranslateRequestMirrorPolicyCluster(GetDestinationCluster(dst, service, listenerPort), mp)
+}
+
+func TranslateRequestMirrorPolicyCluster(cluster string, mp *core.RuntimeFractionalPercent,
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/route/route.go:1394]: why abstact another function

**Review by hzxuzhonghu — COMMENTED** [review]
Should have some new ut coverage

File: pilot/pkg/networking/core/route/route.go (hunk 1 above)
Comment (howardjohn) [reply pilot/pkg/networking/core/route/route.go:1394]: what do you prefer? to just copy the creation into the 3 places we use it?

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/route/route.go:1394]: Maybe i misread it, cannot `TranslateRequestMirrorPolicy` be used

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/route/route.go:1394]: Actually i donot see the diff with/without this patch

Comment (keithmattix) [reply pilot/pkg/networking/core/route/route.go:1394]: Yeah this is extremely confusing

Comment (howardjohn) [reply pilot/pkg/networking/core/route/route.go:1394]: I renamed stuff to be `waypointXYZ`

**Comment by hzxuzhonghu** [comment]
Clear now, LGTM

**Comment by howardjohn** [comment]
relnote added

**Review by stevenctl — APPROVED** [review]
do we have the mirroring test disabled for waypoints? can we enable it?

**Pull request merged**